docs/shared/tutorials/npm-workspaces.md

```diff
@@ -0,0 +1,353 @@
+
+```json {% fileName="nx.json" %}
+{
+  "$schema": "./node_modules/nx/schemas/nx-schema.json",
+  "targetDefaults": {
+    "build": {
+      "dependsOn": ["^build", "typecheck"],
+      "cache": true
+    },
+    "typecheck": {
+      "cache": true
+    },
+    "lint": {
+      "cache": true
+    }
+  },
+  "defaultBase": "main"
+}
+```
+
+Now let's add the `@nx/vite` plugin:
+
+```{% command="npx nx add @nx/vite" path="~/tuskydesign" %}
+✔ Installing @nx/vite@18.1.0...
+✔ Initializing @nx/vite...
+
+ NX   Package @nx/vite added successfully.
+```
+
+The `nx add` command installs the version of the plugin that matches your repo's Nx version and runs that plugin's initialization script. For `@nx/vite`, the initialization script registers the plugin in the `plugins` array of `nx.json` and updates any `package.json` scripts that execute Vite related tasks. Open the project details view for the `demo` app and look at the `build` task.
+
+```shell {% path="~/tuskydesigns" %}
+npx nx show project @tuskdesign/demo --web
+```
+
+{% project-details title="Project Details View" jsonFile="shared/tutorials/npm-workspaces-pdv2.json" %}
+{% /project-details %}
+
+If you hover over the settings for the `build` task, you can see where those settings come from. The `inputs` and `outputs` are defined by the `@nx/vite` plugin from the `vite.config.ts` file where as the `dependsOn` property we set earlier in the tutorial in the `targetDefaults` in the `nx.json` file.
+
+Now let's change where the `build` results are output to in the `vite.config.ts` file.
+
+```ts {% fileName="apps/demo/vite.config.ts" highlightLines=[7-9] %}
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  build: {
+    outDir: '../../dist/demo',
+  },
+});
+```
+
+Now if you look at project details view again, you'll see that the `outputs` property for Nx's caching has been updated to stay in sync with the setting in the `vite.config.ts` file.
+
+You can also add the `@nx/eslint` plugin to see how it infers `lint` tasks based on the ESLint configuration files.
+
+## Summary
+
+After following this tutorial, the repository is still using all the same tools to run tasks, but now Nx runs those tasks in a smarter way. The tasks are efficiently cached so that there is no repeated work and the cache configuration settings are automatically synced with your tooling configuration files by Nx plugins. Also, any task dependencies are automatically executed whenever needed because we configured task pipelines for the projects.
+
+The final task graph for `demo` app's `build` task looks like this:
+
+{% graph height="200px" title="Build Task Pipeline" type="task" jsonFile="shared/tutorials/npm-workspaces-build-tasks2.json" %}
+{% /graph %}
+
+## CI with Nx
```

Comment:
can we link to the CI tutorial? Also link to the recipes for the various CI providers so someone following along with this tutorial would have links to continue